frontend/src/component/tags/TagTypeForm/TagTypeColorPicker.tsx

```diff
@@ -0,0 +1,103 @@
+import type { FC } from 'react';
+import { styled, Box, useTheme } from '@mui/material';
+
+interface ITagTypeColorPickerProps {
+    selectedColor: string;
+    onChange: (color: string) => void;
+}
+
+interface IColorOption {
+    name: string;
+    value: string;
+}
+
+const StyledColorContainer = styled('div')(({ theme }) => ({
+    display: 'flex',
+    flexDirection: 'column',
+    marginBottom: theme.spacing(2),
+    marginTop: theme.spacing(1),
+}));
+
+const StyledColorsWrapper = styled('div')(({ theme }) => ({
+    display: 'flex',
+    flexWrap: 'wrap',
+    gap: theme.spacing(1),
+    marginTop: theme.spacing(1),
+}));
+
+const StyledColorCircle = styled('button')<{
+    $color: string;
+    $selected: boolean;
+}>(({ theme, $color, $selected }) => ({
+    width: '24px',
+    height: '24px',
+    borderRadius: '50%',
+    border: $selected
+        ? `2px solid ${theme.palette.primary.main}`
+        : $color === '#FFFFFF'
+          ? `1px solid ${theme.palette.divider}`
+          : `1px solid ${$color}`,
+    backgroundColor: $color,
+    cursor: 'pointer',
+    padding: 0,
+    '&:hover': {
+        boxShadow: theme.boxShadows.elevated,
+    },
+}));
+
+export const TagTypeColorPicker: FC<ITagTypeColorPickerProps> = ({
+    selectedColor,
+    onChange,
+}) => {
+    const theme = useTheme();
+
+    const getColorWithFallback = (color: string | undefined): string =>
+        color || '#FFFFFF';
+
+    const colorOptions: IColorOption[] = [
+        { name: 'Common white', value: theme.palette.common.white },
+        {
+            name: 'Success border',
```

Comment:
are we using those names in the aria? what does border color has to do with tag color?

Reply:
Names could be more descriptive.